```
 pkg/sql/contention/event_store_test.go:142:24: not enough arguments in call to newEventStore
```

From the failed bors attempts. Is it due to this PR?

bors r-

--- a/docs/generated/settings/settings-for-tenants.txt
+++ b/docs/generated/settings/settings-for-tenants.txt
@@ -86,6 +86,7 @@ server.web_session.purge.ttl	duration	1h0m0s	if nonzero, entries in system.web_s
 server.web_session_timeout	duration	168h0m0s	the duration that a newly created web session will be valid
 sql.auth.resolve_membership_single_scan.enabled	boolean	true	determines whether to populate the role membership cache with a single scan
 sql.contention.event_store.capacity	byte size	64 MiB	the in-memory storage capacity per-node of contention event store
+sql.contention.event_store.duration_threshold	duration	0s	minimum contention duration to cause the contention events to be collected into crdb_internal.transaction_contention_events
 sql.contention.txn_id_cache.max_size	byte size	0 B	the maximum byte size TxnID cache will use (set to 0 to disable)
 sql.cross_db_fks.enabled	boolean	false	if true, creating foreign key references across databases is allowed
 sql.cross_db_sequence_owners.enabled	boolean	false	if true, creating sequences owned by tables from other databases is allowed

--- a/docs/generated/settings/settings.html
+++ b/docs/generated/settings/settings.html
@@ -100,6 +100,7 @@
 <tr><td><code>server.web_session_timeout</code></td><td>duration</td><td><code>168h0m0s</code></td><td>the duration that a newly created web session will be valid</td></tr>
 <tr><td><code>sql.auth.resolve_membership_single_scan.enabled</code></td><td>boolean</td><td><code>true</code></td><td>determines whether to populate the role membership cache with a single scan</td></tr>
 <tr><td><code>sql.contention.event_store.capacity</code></td><td>byte size</td><td><code>64 MiB</code></td><td>the in-memory storage capacity per-node of contention event store</td></tr>
+<tr><td><code>sql.contention.event_store.duration_threshold</code></td><td>duration</td><td><code>0s</code></td><td>minimum contention duration to cause the contention events to be collected into crdb_internal.transaction_contention_events</td></tr>
 <tr><td><code>sql.contention.txn_id_cache.max_size</code></td><td>byte size</td><td><code>0 B</code></td><td>the maximum byte size TxnID cache will use (set to 0 to disable)</td></tr>
 <tr><td><code>sql.cross_db_fks.enabled</code></td><td>boolean</td><td><code>false</code></td><td>if true, creating foreign key references across databases is allowed</td></tr>
 <tr><td><code>sql.cross_db_sequence_owners.enabled</code></td><td>boolean</td><td><code>false</code></td><td>if true, creating sequences owned by tables from other databases is allowed</td></tr>

--- a/pkg/sql/contention/cluster_settings.go
+++ b/pkg/sql/contention/cluster_settings.go
@@ -34,3 +34,14 @@ var StoreCapacity = settings.RegisterByteSizeSetting(
 	"the in-memory storage capacity per-node of contention event store",
 	64*1024*1024, // 64 MB per node.
 ).WithPublic()
+
+// DurationThreshold is the cluster setting for the threshold of
+// contention durations. Only the contention events whose duration exceeds the
+// threshold will be collected into crdb_internal.transaction_contention_events.
+var DurationThreshold = settings.RegisterDurationSetting(
+	settings.TenantWritable,
+	"sql.contention.event_store.duration_threshold",
+	"minimum contention duration to cause the contention events to be collected "+
+		"into crdb_internal.transaction_contention_events",
+	0,
+).WithPublic()

--- a/pkg/sql/contention/event_store.go
+++ b/pkg/sql/contention/event_store.go
@@ -234,9 +234,20 @@ func (s *eventStore) startResolver(ctx context.Context, stopper *stop.Stopper) {
 
 // addEvent implements the eventWriter interface.
 func (s *eventStore) addEvent(e contentionpb.ExtendedContentionEvent) {
+	// Setting the TxnIDResolutionInterval to 0 effectively disables the
+	// eventStore.
 	if TxnIDResolutionInterval.Get(&s.st.SV) == 0 {
 		return
 	}
+
+	// If the duration threshold is set, we only collect contention events whose
+	// duration exceeds the threshold.
+	if threshold := DurationThreshold.Get(&s.st.SV); threshold > 0 {
+		if e.BlockingEvent.Duration < threshold {
+			return
+		}
+	}
+
 	s.guard.AtomicWrite(func(writerIdx int64) {
 		e.CollectionTs = s.timeSrc()
 		s.guard.buffer[writerIdx] = e

--- a/pkg/sql/contention/event_store_test.go
+++ b/pkg/sql/contention/event_store_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql/contentionpb"
+	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
@@ -123,6 +124,71 @@ func TestEventStore(t *testing.T) {
 			require.Equal(t, expectedEvent, *actual)
 			return nil
 		}))
+}
+
+func TestCollectionThreshold(t *testing.T) {
+	ctx := context.Background()
+
+	stopper := stop.NewStopper()
+	defer stopper.Stop(ctx)
+
+	st := cluster.MakeTestingClusterSettings()
+
+	// Only collect contention events that has over 50 ms of contention duration.
+	threshold := 50 * time.Millisecond
+	DurationThreshold.Override(ctx, &st.SV, threshold)
+	statusServer := newFakeStatusServerCluster()
+
+	metrics := NewMetrics()
+	store := newEventStore(st, statusServer.txnIDResolution, time.Now, &metrics)
+	store.start(ctx, stopper)
+
+	input := []contentionpb.ExtendedContentionEvent{
+		{
+			BlockingEvent: roachpb.ContentionEvent{
+				TxnMeta: enginepb.TxnMeta{
+					ID: uuid.FastMakeV4(),
+				},
+				Duration: 10 * time.Millisecond,
+			},
+		},
+		{
+			BlockingEvent: roachpb.ContentionEvent{
+				TxnMeta: enginepb.TxnMeta{
+					ID: uuid.FastMakeV4(),
+				},
+				Duration: 2 * time.Second,
+			},
+		},
+	}
+
+	for i := range input {
+		store.addEvent(input[i])
+	}
+
+	const expectedNumOfEntries = 1
+
+	// Force the contention events to get flushed.
+	testutils.SucceedsWithin(t, func() error {
+		store.guard.ForceSync()
+		numOfEntries := 0
+		require.NoError(t,
+			store.ForEachEvent(func(e *contentionpb.ExtendedContentionEvent) error {
+				require.GreaterOrEqualf(t, e.BlockingEvent.Duration, threshold,
+					"expect contention event's duration to exceed the threshold of %s, "+
+						"but it didn't", threshold)
+				numOfEntries++
+				return nil
+			},
+			))
+
+		if numOfEntries != expectedNumOfEntries {
+			return errors.Newf("expected to have %d contention events, but %d "+
+				"events are found", expectedNumOfEntries, numOfEntries)
+		}
+
+		return nil
+	}, 3*time.Second)
 }
 
 func BenchmarkEventStoreIntake(b *testing.B) {

